lib/puppet/module/task.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/util/logging'
+require 'json'
+
+class Puppet::Module
+  class Task
+    class Error < Puppet::Error; end
+    class InvalidName < Error; end
+    class InvalidFile < Error; end
+    class TaskNotFound < Error; end
+
+    FORBIDDEN_EXTENSIONS = %w{.conf .md}
+
+    def self.is_task_name?(name)
+      return true if name =~ /^[a-z][a-z0-9_]*$/
+      return false
+    end
+
+    # Determine whether a file has a legal name for either a task's executable or metadata file.
+    def self.is_tasks_filename?(path)
+      name_less_extension = File.basename(path, '.*')
+      return false if not is_task_name?(name_less_extension)
+      FORBIDDEN_EXTENSIONS.each do |ext|
+        return false if path.end_with?(ext)
+      end
```

Comment:
not sure if its more readable/idiomatic or not but at least more succinctly this could be done as
```
return !FORBIDDEN_EXTENSIONS.any? { |ext| File.extname(path) == ext }
```